File: clients/desktop/GraNAS.Desktop.App/Services/Api/FoldersApi.cs
using GraNAS.Desktop.Contracts.Metadata;

namespace GraNAS.Desktop.App.Services.Api;

public class FoldersApi : ApiBase, IFoldersApi
{
  public FoldersApi(HttpClient http) : base(http) { }

  public Task<List<FolderResponse>> GetFoldersAsync(CancellationToken ct = default)
    => GetAsync<List<FolderResponse>>("api/metadata/folders", ct);

  public Task<FolderResponse> CreateFolderAsync(CreateFolderRequest request, CancellationToken ct = default)
    => PostAsync<FolderResponse>("api/metadata/folders", request, ct);

  public Task DeleteFolderAsync(Guid id, CancellationToken ct = default)
    => DeleteAsync($"api/metadata/folders/{id}", ct);
}

File: clients/web/src/api/folders.api.ts
import type { CreateFolderRequest, FolderResponse } from '../types/folder';
import { api } from './client';

export const foldersApi = {
  list: () =>
    api.get<FolderResponse[]>('/api/metadata/folders'),

  create: (data: CreateFolderRequest) =>
    api.post<FolderResponse>('/api/metadata/folders', data),

  delete: (id: string) =>
    api.delete(`/api/metadata/folders/${id}`),

  touch: (id: string) =>
    api.patch(`/api/metadata/folders/${id}/touch`),
};
